packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -0,0 +1,290 @@
+
+## Background TODO - description from V8
+
+A SwatchPicker displays color, image and pattern options as a grid. It can be shown by itself, with a header and dividers, or as a button that expands to show the SwatchPicker.
+
+## Prior Art
+
+- [Convergence epic](https://github.com/microsoft/fluentui/issues/28606)
+
+### Fabric (v8)
+
+```jsx
+import { IColorCellProps, SwatchColorPicker } from '@fluentui/react/lib/SwatchColorPicker';
+
+const colorCellsExample1 = [
+  { id: 'a', label: 'orange', color: '#ca5010' },
+  { id: 'b', label: 'cyan', color: '#038387' },
+  { id: 'c', label: 'blueMagenta', color: '#8764b8' },
+  { id: 'd', label: 'magenta', color: '#881798' },
+  { id: 'e', label: 'white', color: '#ffffff' },
+];
+
+export const SwatchColorPickerBasicExample: React.FunctionComponent = () => {
+  const [previewColor, setPreviewColor] = React.useState<string>();
+  const baseId = useId('colorpicker');
+
+  const swatchColorPickerOnCellHovered = (id: string, color: string) => {
+    setPreviewColor(color!);
+  };
+
+  return (
+    <>
+      <div id={`${baseId}-circle`}>Simple circle swatch color picker:</div>
+      <SwatchColorPicker
+        columnCount={5}
+        cellShape={'circle'}
+        colorCells={colorCellsExample1}
+        aria-labelledby={`${baseId}-circle`}
+      />
+      <div id={`${baseId}-square`}>Simple square swatch color picker with default size of 20px:</div>
+      <SwatchColorPicker
+        columnCount={5}
+        cellShape={'square'}
+        colorCells={colorCellsExample1}
+        aria-labelledby={`${baseId}-square`}
+      />
+      <div id={`${baseId}-custom-size`}>Simple square swatch color picker with custom size of 35px:</div>
+      <SwatchColorPicker
+        columnCount={5}
+        cellHeight={35}
+        cellWidth={35}
+        cellShape={'square'}
+        colorCells={colorCellsExample1}
+        aria-labelledby={`${baseId}-custom-size`}
+      />
+    </>
+  );
+};
+```
+
+### 3rd party Design Systems
+
+- Adobe Spectrum
+  - [SwatchGroup](https://spectrum.adobe.com/page/swatch-group/)
+  - [Swatch](https://spectrum.adobe.com/page/swatch/)
+
+### Components
+
+| Purpose                                                      | Fabric (V8)         | V9           | Matching? |
+| ------------------------------------------------------------ | ------------------- | ------------ | --------- |
+| Component responsible for rendering swatches as row and grid | SwatchColorPicker   | SwatchPicker | ⚠️        |
+| Color cell                                                   | ColorPickerGridCell | ColorSwatch  | ⚠️        |
+| Image cell                                                   |                     | ImageSwatch  | ❌        |
+
+## Sample Code
+
+```jsx
+<SwatchPicker
+  layout="grid"
+  swatch={color}
+  onChange={(\_, data) => setSwatch(data.value)}
+  aria-labelledby="colors"
+  name="color"
+  style={{ gridTemplateColumns: `repeat(4, 30px)` }} >
+    <ColorSwatch id={0} swatch="red" />
```

Comment:
Usage of DOM IDs does not seem to be a good idea as they should be unique per a document. Please consider something different.

Reply:
What does the `id` prop represent here? If it's a DOM id then I'd agree with @layershifter. If not, it should be a simple exercise to rename the prop to something else so it doesn't clash with the DOM attribute

Reply:
It shouldn't be a DOM id, it was sort of `swatchId`. I'm still thinking how to manage it. One of ideas was to pass `swatchId` which will represent color/pattern/image